codice/ddf
# DDF-05129 Range search for numeric values

#### What does this PR do?
Adds a "between" filter for number ranges - effectively a shortcut for value1 <= x <= value2. 
#### Who is reviewing it? 
@andrewkfiedler 
@rzwiefel 
@jordanwilking 
@Bdthomson 
@glenhein 

#### Issues Addressed
#5129

#### Screenshots
<img width="772" alt="Screen Shot 2019-08-13 at 12 23 32 PM" src="https://user-images.githubusercontent.com/51682089/62970841-3b250680-bdc5-11e9-971d-71fdadfecbba.png">





#### How should this be tested?
The range filter can be tested by querying based off of numeric attributes. For example:
1. Upload a file to create a test resource
2. Add the location.altitude-meters attribute to the uploaded resource and set any value
3. Add the media.number-of-bands attribute to the uploaded resource and set any value
4. Perform a query against the location.altitude-meters attribute and use the "RANGE" selection for the comparator.
5. Perform the search with a range that includes the value from step 2 and one that does not include it. Validate that the resource appears when included and does not when excluded.
6. Repeat the same process as step 5, but using the media.number-of-bands field - this time try to use a non-integer value and verify that an error indicates the value is of the wrong type.

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
